### DISCUSSION TITLE
v20.2.0

### DISCUSSION THREAD

**Discussion by fballiano** [OpenMage/magento-lts#3617] [Releases]
OpenMage continues its remarkable journey with an important new release!
Among the numerous pull requests, one truly stands out. After months of development, we are thrilled to announce a significant upgrade to a core component of the OpenMage backend: the tinyMCE rich text editor skyrockets from its "more than 10-years-old" version 3 to the new and shining version 6.7!

## Highlights of OpenMage 20.2.0
📝 **TinyMCE is updated from v3 to v6.7.1**
🔌 composer.json now allows **PHP 8.3**
🧹 Added a cron job to **clean expired API sessions**
📊 Added "googleanalytics_ga4_send_data_before" to **customize GA4 data**
🚀 Improved **rate limit functions**
🛠️ Fixed "Credit Memo not created when refund issued by merchant"
📄 Switched DOCTYPE for base/adminhtml themes to HTML5

## Disclaimer

Since this is another substantial update please be sure to backup and test before deploying to production!

## Changelog
* Added a cron job to clean expired API sessions by @kiatng in https://github.com/OpenMage/magento-lts/pull/3479
* Fixed incompatible interface deprecations in DB Session Adapter by @elidrissidev in https://github.com/OpenMage/magento-lts/pull/3497
* Fixed PHPCS github workflow for unexisting ruleset by @sreichel in https://github.com/OpenMage/magento-lts/pull/3500
* Fixed some PHPCS errors by @fballiano in https://github.com/OpenMage/magento-lts/pull/3501
* Added "googleanalytics_ga4_send_data_before" to customize GA4 data by @fballiano in https://github.com/OpenMage/magento-lts/pull/3448
* Bump friendsofphp/php-cs-fixer from 3.25.1 to 3.26.1 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3506
* Bump phpstan/phpstan from 1.10.32 to 1.10.33 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3505
* Bump tj-actions/changed-files from 38 to 39 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3504
* Bump actions/checkout from 3 to 4 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3503
* Improved rate limit functions by @Judx in https://github.com/OpenMage/magento-lts/pull/3507
* Replaced Magento placeholder images with OpenMage ones by @empiricompany in https://github.com/OpenMage/magento-lts/pull/3513
* Fixed check if ConfigurableSwatches is enabled by @luigifab in https://github.com/OpenMage/magento-lts/pull/2727
* Fixed docblock in phtml in catalog/product/view/options/type by @kiatng in https://github.com/OpenMage/magento-lts/pull/3517
* Fixed deprecated functionality str_contains() in Design/Package.php by @fballiano in https://github.com/OpenMage/magento-lts/pull/3516
* Bump friendsofphp/php-cs-fixer from 3.26.1 to 3.27.0 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3521
* Bump phpstan/phpstan from 1.10.33 to 1.10.34 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3524
* Bump phpunit/phpunit from 9.6.11 to 9.6.12 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3522
* Bump phpseclib/phpseclib from 3.0.21 to 3.0.22 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3523
* Link to NotifyStock feed is shown only if enabled by @fballiano in https://github.com/OpenMage/magento-lts/pull/3520
* Deprecated decorate*() js functions by @fballiano in https://github.com/OpenMage/magento-lts/pull/3526
* Updated composer.json to allow PHP 8.3 by @kyrena in https://github.com/OpenMage/magento-lts/pull/3536
* Fixed passing null for explode is deprecated in Mage_Adminhtml_Block_System_Currency_Rate_Matrix by @kyrena in https://github.com/OpenMage/magento-lts/pull/3539
* Bump phpunit/phpunit from 9.6.12 to 9.6.13 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3547
* Bump phpstan/phpstan from 1.10.34 to 1.10.35 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3549
* Bump friendsofphp/php-cs-fixer from 3.27.0 to 3.28.0 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3550
* Bump phpseclib/phpseclib from 3.0.22 to 3.0.23 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3548
* Removed some weirdly formatted comments by @fballiano in https://github.com/OpenMage/magento-lts/pull/3552
* Fixed empty exception when customer does not exist for products alerts by @kyrena in https://github.com/OpenMage/magento-lts/pull/3537
* Use a core/flag to store tax notification ignored flag rather than co… by @colinmollenhour in https://github.com/OpenMage/magento-lts/pull/3532
* Fixed PHPStan workflows not running on the correct PHP version by @fballiano in https://github.com/OpenMage/magento-lts/pull/3542
* Recovered some lost deprecated notice in #3552 by @fballiano in https://github.com/OpenMage/magento-lts/pull/3558
* Bump friendsofphp/php-cs-fixer from 3.28.0 to 3.34.0 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3565
* Bump phpstan/phpstan from 1.10.35 to 1.10.36 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3566
* Bump phpmd/phpmd from 2.13.0 to 2.14.1 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3567
* Fixed docblock onepage.phtml, cart.phtml, and success.phtml in checkout. by @kiatng in https://github.com/OpenMage/magento-lts/pull/3568
* Excluded Varien_Data_Form_Element_Label from "edit attributes" mass action by @Hanmac in https://github.com/OpenMage/magento-lts/pull/3540
* Fixed updated_at doesn't update on mass actions by @Tomasz-Silpion in https://github.com/OpenMage/magento-lts/pull/3529
* Minor fixes for TinyMCE v6 by @fballiano in https://github.com/OpenMage/magento-lts/pull/3570
* Forced PHP version to 7.4 for PHPCS related workflows by @fballiano in https://github.com/OpenMage/magento-lts/pull/3571
* Reinit fresh config before flushing cache and immediately save config… by @colinmollenhour in https://github.com/OpenMage/magento-lts/pull/3533
* Fixed "Credit Memo not created when refund issued by merchant" by @luigifab in https://github.com/OpenMage/magento-lts/pull/2729
* Fixed PHP 8.2 warnings in Cm_RedisSession (updated to 3.2.0) by @colinmollenhour in https://github.com/OpenMage/magento-lts/pull/3576
* Switched DOCTYPE to HTML5 for base theme by @fballiano in https://github.com/OpenMage/magento-lts/pull/3574
* Switched DOCTYPE to HTML5 for adminhtml themes by @fballiano in https://github.com/OpenMage/magento-lts/pull/3579
* Removed text-shadow from disabled dropdown in Admin by @elidrissidev in https://github.com/OpenMage/magento-lts/pull/3583
* Bump phpstan/phpstan from 1.10.36 to 1.10.38 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3586
* Bump friendsofphp/php-cs-fixer from 3.34.0 to 3.34.1 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3587
* Add status column to customer orders grid in Admin by @elidrissidev in https://github.com/OpenMage/magento-lts/pull/3582
* Fix syntax in Mage_Adminhtml.csv by @elidrissidev in https://github.com/OpenMage/magento-lts/pull/3590
* Fixed some phtml docblock in folder adminhtml/default/default by @kiatng in https://github.com/OpenMage/magento-lts/pull/3588
* Bump friendsofphp/php-cs-fixer from 3.34.1 to 3.35.1 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3597
* bugfix/LogVisitorSaveByRequest: added initServerData() to populate ServerData on saving visitor_log information by @eneiasramos in https://github.com/OpenMage/magento-lts/pull/3598
* Exception is now printed on Mage_Catalog_ProductController::viewAction() in developer mode by @kyrena in https://github.com/OpenMage/magento-lts/pull/3578
* Bump phpstan/phpstan from 1.10.38 to 1.10.39 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3603
* Bump phpseclib/phpseclib from 3.0.23 to 3.0.33 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3602
* Bump pelago/emogrifier from 7.0.0 to 7.1.0 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3601
* Updated TinyMCE to 6.7.1 by @fballiano in https://github.com/OpenMage/magento-lts/pull/3605
* Fixed implicit conversion to int is deprecated in Mage_Shell_Indexer by @luigifab in https://github.com/OpenMage/magento-lts/pull/3612
* Fixed bug on incorrect join in addAttributeToSelect() when the second parameter is true by @kiatng in https://github.com/OpenMage/magento-lts/pull/3609

**Full Changelog**: https://github.com/OpenMage/magento-lts/compare/v20.1.1...v20.2.0

<hr /><em>This discussion was created from the release <a href='https://github.com/OpenMage/magento-lts/releases/tag/v20.2.0'>v20.2.0</a>.</em>

**Comment by m-overlund** [comment]
> ## Highlights of OpenMage 20.1.1

Version number needs to be changed 😊

**Reply by fballiano** [reply]
when you check 100 times and you don't see the biggest font of the page 😅

**Reply by m-overlund** [reply]
yeah, i know the feeling 😅
Great job though 🚀

**Reply by luigifab** [reply]
It seems, that we are almost deprecated free with PHP 8.1 / 8.2 / 8.3!
Even if I don't use all functionality of OpenMage.

**Comment by empiricompany** [comment]
🚀 ready for my next projects! thanks @fballiano and this amazing community!
